components/copy_url_context_menu/copy_url_context_menu.tsx

```diff
@@ -1,30 +1,35 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {ContextMenu, ContextMenuTrigger, MenuItem} from 'react-contextmenu';
 import {FormattedMessage} from 'react-intl';
 
-export default class CopyUrlContextMenu extends React.PureComponent {
-    static propTypes = {
+type Props = {
 
-        // The child component that will be right-clicked on to show the context menu
-        children: PropTypes.element,
+    /**
+     * The child component that will be right-clicked on to show the context menu
+     */
+    children: React.ReactNode;
 
-        // The link to copy to the user's clipboard when the 'Copy' option is selected from the context menu
-        link: PropTypes.string.isRequired,
+    /**
+     * The link to copy to the user's clipboard when the 'Copy' option is selected from the context menu
+     */
+    link: string;
 
-        // A unique id differentiating this instance of context menu from others on the page.
-        menuId: PropTypes.string.isRequired,
+    /**
+     * A unique id differentiating this instance of context menu from others on the page
+     */
+    menuId: string;
 
-        siteURL: PropTypes.string.isRequired,
+    siteURL?: string;
 
-        actions: PropTypes.shape({
-            copyToClipboard: PropTypes.func.isRequired,
-        }),
+    actions: {
+        copyToClipboard: (link: string) => void;
     };
+}
 
+export default class CopyUrlContextMenu extends React.PureComponent<Props> {
```

Comment:
Hey @LauSam09 . Thanks a lot for this good work. We really appreciate you helping us out. Everything looks good and I just have a small nit/request, Is it possible to also type the return of the render() function? ```render(): JSX.Elelment``` :)

Reply:
Hello @AGMETEOR, you're welcome! Yeah of course, I'll add them now. I looked at previous PRs and didn't see any examples of people defining return types for functions which seemed weird given that `eslint` was very unhappy about it!

Reply:
@AGMETEOR I've taken your comment literally and added a return type just for the `render` method - if this was right I'm curious why the other functions shouldn't have return types defined?

Reply:
Sorry if I made some confusion. Atm, the eslint rules don't enforce it. Hopefully, I have answered your question :)